addons/website_sale/models/sale_order.py

```diff
@@ -22,6 +22,12 @@ class SaleOrder(models.Model):
     payment_acquirer_id = fields.Many2one('payment.acquirer', string='Payment Acquirer', copy=False)
     payment_tx_id = fields.Many2one('payment.transaction', string='Transaction', copy=False)
     only_services = fields.Boolean(compute='_compute_cart_info', string='Only Services')
+    is_public_user = fields.Boolean(string="Is Public User", compute='_compute_is_public_user', store=True)
```

Comment:
@JKE-be , yes you are right, no need to add new field.

and we need to eval domain in client side due to datetime checking
so we use eval in context and pass id of public user partner.